src/measure_theory/function/conditional_expectation.lean

```diff
@@ -2043,6 +2085,136 @@ end
+begin
+  have hsf_zero : ∀ g : α → F', g =ᵐ[μ.restrict sᶜ] 0 → s.indicator g =ᵐ[μ] g,
+    from λ g, indicator_ae_eq_of_restrict_compl_ae_eq_zero (hm _ hs),
+  refine (hsf_zero (μ[f | m]) (condexp_ae_eq_restrict_zero hs.compl hf)).trans _,
+  exact condexp_congr_ae (hsf_zero f hf).symm,
+end
+
+lemma condexp_indicator (hf_int : integrable f μ) (hs : measurable_set[m] s) :
+  μ[s.indicator f | m] =ᵐ[μ] s.indicator (μ[f | m]) :=
+begin
+  by_cases hm : m ≤ m0,
+  swap, { simp_rw [condexp_of_not_le hm, set.indicator_zero'], },
+  by_cases hμm : sigma_finite (μ.trim hm),
+  swap, { simp_rw [condexp_of_not_sigma_finite hm hμm, set.indicator_zero'], },
+  haveI : sigma_finite (μ.trim hm) := hμm,
+  -- use `have` to perform what should be the first calc step because of an error I don't
+  -- understand
+  have : s.indicator (μ[f|m]) =ᵐ[μ] s.indicator (μ[s.indicator f + sᶜ.indicator f|m]),
+    by rw set.indicator_self_add_compl s f,
+  refine (this.trans _).symm,
+  calc s.indicator (μ[s.indicator f + sᶜ.indicator f|m])
+      =ᵐ[μ] s.indicator (μ[s.indicator f|m] + μ[sᶜ.indicator f|m]) :
+    begin
+      have : μ[s.indicator f + sᶜ.indicator f|m] =ᵐ[μ] μ[s.indicator f|m] + μ[sᶜ.indicator f|m],
+        from condexp_add (hf_int.indicator (hm _ hs)) (hf_int.indicator (hm _ hs.compl)),
+      filter_upwards [this] with x hx,
+      classical,
+      rw [set.indicator_apply, set.indicator_apply, hx],
+    end
+  ... = s.indicator (μ[s.indicator f|m]) + s.indicator (μ[sᶜ.indicator f|m]) :
+    s.indicator_add' _ _
+  ... =ᵐ[μ] s.indicator (μ[s.indicator f|m]) + s.indicator (sᶜ.indicator (μ[sᶜ.indicator f|m])) :
+    begin
+      refine filter.eventually_eq.rfl.add _,
+      have : sᶜ.indicator (μ[sᶜ.indicator f|m]) =ᵐ[μ] μ[sᶜ.indicator f|m],
+      { refine (condexp_indicator_aux hm hs.compl _).trans _,
+        { exact indicator_ae_eq_restrict_compl (hm _ hs.compl), },
+        { rw [set.indicator_indicator, set.inter_self], }, },
+      filter_upwards [this] with x hx,
+      by_cases hxs : x ∈ s,
+      { simp only [hx, hxs, set.indicator_of_mem], },
+      { simp only [hxs, set.indicator_of_not_mem, not_false_iff], },
+    end
+  ... =ᵐ[μ] s.indicator (μ[s.indicator f|m]) :
+    by rw [set.indicator_indicator, set.inter_compl_self, set.indicator_empty', add_zero]
+  ... =ᵐ[μ] μ[s.indicator f|m] :
+    begin
+      refine (condexp_indicator_aux hm hs _).trans _,
+      { exact indicator_ae_eq_restrict_compl (hm _ hs), },
+      { rw [set.indicator_indicator, set.inter_self], },
+    end
+end
+
+/-- If the restriction to a `m`-measurable set `s` of a σ-algebra `m` is equal to the restriction
+to `s` of another σ-algebra `m₂` (hypothesis `hs`), then `μ[f | m] =ᵐ[μ.restrict s] μ[f | m₂]`. -/
+lemma condexp_ae_eq_restrict_of_measurable_space_eq_on {m m₂ m0 : measurable_space α}
+  {μ : measure α}
+  (hm : m ≤ m0) (hm₂ : m₂ ≤ m0) [sigma_finite (μ.trim hm)] [sigma_finite (μ.trim hm₂)]
+  (hf_int : integrable f μ) (hs_m : measurable_set[m] s)
```

Comment:
Do you really need `hf_int` here? Otherwise, the conditional expectations should be zero, so the conclusion should still be true.